File: linkedList/206. Reverse Linked List/main.py
class Solution(object):
    
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        # [ None , 1 , 2, None]

        prev = None
        cur = head
        while cur != None:
           
            head = head.next
            cur.next = prev
            prev = cur
            cur = head
        return prev


            
            
            
        

        
          

            

            
   

        

        

        
        
        
    
        

File: linkedList/206. Reverse Linked List/C/main.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    
    if (head == NULL || head -> next == NULL)
        return head;

    // declare the pre Node 
    struct ListNode *preNode = NULL;

    // declare the cur Node , 
    struct ListNode *curNode = head ;

    // initial the nextnode 
    struct ListNode *nextNode ;

    while (curNode != NULL){

        nextNode = curNode -> next;
        curNode -> next = preNode;
        preNode = curNode;
        curNode = nextNode;
        
    }

    head = preNode;
    return preNode;
}
